### PR TITLE
 Backend: more efficient implementation of String.removeColor() method (StringUtils.kt:)

### PR DESCRIPTION
**Pull request by brainage04**
Tiny optimization, will reduce the number of loop iterations in `String.removeColor()` by `n`, where `n` is the number of formatting characters (`§`) in the string being formatted.

Resubmitted with more appropriate branch name.

### CODE REVIEW

**Comment by hannibal002** [comment]
I have tested this optimization: ([Test code](https://gist.github.com/hannibal002/44086924bc0c236c06e1e4f1421b56ec))



After 350k runs with real life data:
```
total time: '89.416600ms' vs '58.966100ms' vs '325.373200ms' vs '419.861600ms'
testLog.log("total time: '$totalOld' vs '$totalNew' vs '$totalRegexCached' vs '$totalRegexUncached'")
```

image
Interesting results indeed.

**Comment by hannibal002** [comment]
https://gist.github.com/hannibal002/fe5f04991841b0f2e00cfaa0b101241b
I changed the logging process in this example and added the number of runs:
```
numberOfRuns: 346963 = '73.690300ms' vs '101.772400ms' vs '281.423500ms' vs '420.686900ms'
```

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt [hunk 1]
```diff
@@ -19,20 +19,17 @@ object StringUtils {
...
-                continue
-            }
-            if (skipNext) {
-                skipNext = false
+
+        var counter = 0
+        while (counter < this.length) {
+            if (this[counter] == '§') {
+                counter += 2
                 continue
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt:28]: I think that continue can be removed

**Review by lineargraph — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt [hunk 2]
```diff
@@ -19,20 +19,17 @@ object StringUtils {
     }
 
     fun String.removeColor(): String {
-//        return replace("(?i)\\u00A7.", "")
-
         val builder = StringBuilder()
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/utils/StringUtils.kt:22]: Also: we can initialize this StringBuilder with a capacity
```suggestion
        val builder = StringBuilder(this.length)
```

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**